### PR TITLE
Feat : Blog - Get All Blog Posts

### PR DESCRIPTION
**Pull request by Waptech07**
This PR contains the implementation of ```[GET] /api/v1/blogs``` [UNPROTECTED Endpoint].
​

## Description

Added functionality for retrieving all blog posts. This includes creating a new endpoint /api/v1/blogs that fetches all blog posts from the database. The endpoint returns a list of blog posts, each containing details such as title, content, author, tags, etc.
​

## Related Issue (Link to issue ticket)
This is an implementation of an open issue [#243](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/243)

​

## Motivation and Context

This change is required to provide an endpoint for clients to retrieve all blog posts. It enhances the functionality of the application by allowing users to view all the blogs.

​

#### The new functionality has been tested using pytest, FastAPI test client, Postman and manual testing. Tests include:
- Retrieving all blog posts when the database is empty.
- Retrieving all blog posts when there are existing blog entries in the database.

#### Test environment:

- PostgreSQL database for storing blog information.
- Mock database setup using pytest for isolated testing.
- Test client setup using FastAPI for API endpoint testing.

​

## Screenshots (if appropriate - Postman, etc):

​### Successful Response - Postman
image

### Fast API
image

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by Goketech — DISMISSED** [review]
Nice work!

File: api/v1/routes/blog.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+from sqlalchemy.orm import Session
+from typing import List
+
+from api.v1.models.blog import Blog
+from api.v1.schemas.blog import BlogResponse
+from api.db.database import get_db
+
+blog = APIRouter(tags=["Blog"])
+
+@blog.get("/blogs", response_model=List[BlogResponse])
```
Comment (Dev-wonderful) [inline api/v1/routes/blog.py:11]: add the `/blogs` `prefix` to the `APIRouter`  then the get route can be `/`, so everyone using this router can have `/blogs` by default

File: tests/v1/get_all_blogs_test.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    response = client.get("/api/v1/blogs")
+
+    # Assert the response
+    assert response.status_code == 200
+    assert response.json() == []
+
+def test_get_all_blogs_with_data(client, db_session_mock):
+    blog_id = uuid4()
+    author_id = uuid4()
+    timezone_offset = -8.0
```
Comment (Dev-wonderful) [inline tests/v1/get_all_blogs_test.py:39]: Isn't all the `id` using `uuid7`?

**Review by Dev-wonderful — CHANGES_REQUESTED** [review]

**Comment by Waptech07** [comment]
@Dev-wonderful I've made changes

**Comment by Dev-wonderful** [comment]
> @Dev-wonderful I've made changes

What of the `prefix`

File: api/v1/routes/blog.py (hunk 1 above)
Comment (Waptech07) [reply api/v1/routes/blog.py:11]: But the other endpoints has prefix blog instead of blogs

**Comment by Waptech07** [comment]
The other endpoints need blog prefix not blogs, should i still change it.

File: api/v1/routes/blog.py (hunk 1 above)
Comment (Waptech07) [reply api/v1/routes/blog.py]: @Dev-wonderful The change have been made

**Pull request merged**